app/assets/javascripts/map.js

```diff
@@ -1,7 +1,259 @@
+class GoogleMap {
+    constructor({ center, dom_element, zoom }) {
+        this.markers = [];
+        this.infowindow = new google.maps.InfoWindow({ content: "" });
+        this.map = new google.maps.Map(dom_element, {
+            center: new google.maps.LatLng(center.lat, center.lng),
+            scrollwheel: true,
+            zoom: zoom || 13,
+            maxZoom: 17
+        });
+    }
+
+    add_marker({ location, title, icon, description, link }) {
+        var marker = new google.maps.Marker({
+            map: this.map,
+            position: { lat: parseFloat(location.lat), lng: parseFloat(location.lng)},
+            title,
+            icon: { url: icon, scaledSize: new google.maps.Size(35, 35) }
```

Comment:
`icon` can be undefined, in which case an error is raised in the developer console:
```
InvalidValueError: setIcon: not a string; and should be a PinView; and no url property; and no path property
```

Maybe don't pass the icon config into the Marker constructor if `icon` undefined